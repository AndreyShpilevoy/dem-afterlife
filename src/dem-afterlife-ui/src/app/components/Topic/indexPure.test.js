/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0, fp/no-mutation:0 , react/jsx-filename-extension:0 */

import React from 'react';
import {shallow} from 'enzyme';
import {TopicPure} from './index';

jest.mock('components/Avatar');
jest.mock('components/UserName');

const styles = {
    '.default-Topic-parentForumTitleStyle:hover': 'default-Topic-parentForumTitleStyle:hover',
    '.default-Topic-parentForumTitleStyle:visited': 'default-Topic-parentForumTitleStyle:visited',
    '.default-Topic-parentForumTitleStyle:visited:hover': 'default-Topic-parentForumTitleStyle:visited:hover',
    '.default-Topic-wrapper:last-child': 'default-Topic-wrapper:last-child',
    '.default-Topic-wrapper:last-child>.default-Topic-separator': 'default-Topic-wrapper:last-child>.default-Topic-separator',
    centerRow: 'default-Topic-centerRow',
    heightFull: 'default-Topic-heightFull',
    lastPostInfoStyle: 'default-Topic-lastPostInfoStyle',
    noWrap: 'default-Topic-noWrap',
    parentForumTitleStyle: 'default-Topic-parentForumTitleStyle',
    parentForumWrapper: 'default-Topic-parentForumWrapper',
    separator: 'default-Topic-separator',
    textSmallMd: 'default-Topic-textSmallMd',
    titleContainer: 'default-Topic-titleContainer',
    titleStyle: 'default-Topic-titleStyle',
    wrapper: 'default-Topic-wrapper'
};

describe('TopicPure', () => {
    it('component match expected snapshot without parent forum information', () => {
        const topic = {
            id: 1,
            forumId: 1,
            title: 'Как деактивировать бомбу',
            postsCount: 215,
            topicViewsCount: 1315,
            lastPostInfo: {
                timeCreation: new Date('2016-09-19T10:42:32.000Z'),
                authorId: 4,
                authorName: 'Buba',
                authorAvatar: 'http://i70.fastpic.ru/big/2015/0628/36/ccbb1e2cb8ba8dbd379a6a12dc6b8336.jpg',
                authorGroupColor: '#00AA00'
            }
        };
        expect(shallow(<TopicPure className={'topicClassName'} styles={styles} topic={topic} />)).toMatchSnapshot();
    });

    it('component match expected snapshot with parent forum information', () => {
        const topic = {
            id: 1,
            forumId: 1,
            title: 'Как деактивировать бомбу',
            postsCount: 215,
            topicViewsCount: 1315,
            lastPostInfo: {
                timeCreation: new Date('2016-09-19T10:42:32.000Z'),
                authorId: 4,
                authorName: 'Buba',
                authorAvatar: 'http://i70.fastpic.ru/big/2015/0628/36/ccbb1e2cb8ba8dbd379a6a12dc6b8336.jpg',
                authorGroupColor: '#00AA00'
            },
            parentForumId: 10,
            parentForumTitle: 'Общие вопросы'
        };
        expect(shallow(<TopicPure className={'topicClassName'} styles={styles} topic={topic} />)).toMatchSnapshot();
    });
});