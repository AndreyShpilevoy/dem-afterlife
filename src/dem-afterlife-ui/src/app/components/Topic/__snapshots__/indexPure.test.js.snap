// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicPure component match expected snapshot with parent forum information 1`] = `
<div
  className="default-Topic-wrapper"
>
  <withStyles(RowPure)
    className=""
    reverse={false}
  >
    <withStyles(ColumnPure)
      className=""
      lg=""
      lgOffset=""
      md=""
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs={12}
      xsOffset=""
    >
      <withStyles(RowPure)
        className=""
        reverse={false}
      >
        <withStyles(ColumnPure)
          className=""
          lg={9}
          lgOffset=""
          md={7}
          mdOffset=""
          sm=""
          smOffset=""
          xl=""
          xlOffset=""
          xs={12}
          xsOffset=""
        >
          <withStyles(RowPure)
            className="default-Topic-centerRow"
            reverse={false}
          >
            <withStyles(ColumnPure)
              className="default-Topic-titleContainer"
              lg={8}
              lgOffset=""
              md=""
              mdOffset=""
              sm=""
              smOffset=""
              xl=""
              xlOffset=""
              xs={12}
              xsOffset=""
            >
              <withStyles(RowPure)
                className=""
                reverse={false}
              >
                <withStyles(ColumnPure)
                  className=""
                  lg=""
                  lgOffset=""
                  md=""
                  mdOffset=""
                  sm=""
                  smOffset=""
                  xl=""
                  xlOffset=""
                  xs={12}
                  xsOffset=""
                >
                  <withStyles(LinkPure)
                    className="default-Topic-titleStyle"
                    style={Object {}}
                    to="/Topic/1"
                  >
                    Как деактивировать бомбу
                  </withStyles(LinkPure)>
                </withStyles(ColumnPure)>
              </withStyles(RowPure)>
              <withStyles(RowPure)
                className="default-Topic-textSmallMd"
                reverse={false}
              >
                <withStyles(ColumnPure)
                  className=""
                  lg={12}
                  lgOffset=""
                  md={7}
                  mdOffset=""
                  sm=""
                  smOffset=""
                  xl=""
                  xlOffset=""
                  xs={12}
                  xsOffset=""
                >
                  <span
                    className="default-Topic-parentForumWrapper"
                  >
                    <Connect(Term)
                      spaceAfter={true}
                      term={
                        Object {
                          "id": 25,
                          "value": "Forum:",
                        }
                      }
                      untermedPostfix=":"
                    />
                    <withStyles(LinkPure)
                      className="default-Topic-parentForumTitleStyle"
                      style={Object {}}
                      to="/Forum/10"
                    >
                      Общие вопросы
                    </withStyles(LinkPure)>
                  </span>
                </withStyles(ColumnPure)>
                <withStyles(ColumnPure)
                  className=""
                  lg=""
                  lgOffset=""
                  md={5}
                  mdOffset=""
                  sm=""
                  smOffset=""
                  xl=""
                  xlOffset=""
                  xs=""
                  xsOffset=""
                >
                  <withStyles(HiddenPure)
                    className="default-Topic-noWrap"
                    lg="up"
                    md=""
                    sm="down"
                    xl=""
                    xs=""
                  >
                    <Connect(Term)
                      spaceAfter={true}
                      term={
                        Object {
                          "id": 2,
                          "value": "Posts",
                        }
                      }
                      untermedPostfix=":"
                    />
                    215
                  </withStyles(HiddenPure)>
                </withStyles(ColumnPure)>
              </withStyles(RowPure)>
            </withStyles(ColumnPure)>
            <withStyles(ColumnPure)
              className="default-Topic-centerRow"
              lg={2}
              lgOffset=""
              md=""
              mdOffset=""
              sm=""
              smOffset=""
              xl=""
              xlOffset=""
              xs=""
              xsOffset=""
            >
              <withStyles(HiddenPure)
                className=""
                lg=""
                md="down"
                sm=""
                xl=""
                xs=""
              >
                215
              </withStyles(HiddenPure)>
            </withStyles(ColumnPure)>
            <withStyles(ColumnPure)
              className="default-Topic-centerRow"
              lg={2}
              lgOffset=""
              md=""
              mdOffset=""
              sm=""
              smOffset=""
              xl=""
              xlOffset=""
              xs=""
              xsOffset=""
            >
              <withStyles(HiddenPure)
                className=""
                lg=""
                md="down"
                sm=""
                xl=""
                xs=""
              >
                1315
              </withStyles(HiddenPure)>
            </withStyles(ColumnPure)>
          </withStyles(RowPure)>
        </withStyles(ColumnPure)>
        <withStyles(ColumnPure)
          className=""
          lg={3}
          lgOffset=""
          md={5}
          mdOffset=""
          sm=""
          smOffset=""
          xl=""
          xlOffset=""
          xs={12}
          xsOffset=""
        >
          <withStyles(RowPure)
            className="default-Topic-heightFull"
            reverse={false}
          >
            <withStyles(ColumnPure)
              className="default-Topic-lastPostInfoStyle"
              lg={10}
              lgOffset=""
              md=""
              mdOffset=""
              sm=""
              smOffset=""
              xl=""
              xlOffset=""
              xs={12}
              xsOffset=""
            >
              <UserName
                className=""
                color="#00AA00"
                id={4}
                name="Buba"
              />
              <withStyles(HiddenPure)
                className=""
                lg="up"
                md=""
                sm=""
                xl=""
                xs=""
              >
                <Connect(Term)
                  doNotTerm={true}
                  indentAfter={true}
                  indentBefore={true}
                  spaceAfter={true}
                  spaceBefore={true}
                  untermedPostfix="-"
                />
              </withStyles(HiddenPure)>
              <Connect(RelativeDateTime)
                relativeDateTime={2016-09-19T10:42:32.000Z}
              />
            </withStyles(ColumnPure)>
            <withStyles(ColumnPure)
              className=""
              lg={2}
              lgOffset=""
              md=""
              mdOffset=""
              sm=""
              smOffset=""
              xl=""
              xlOffset=""
              xs=""
              xsOffset=""
            >
              <withStyles(HiddenPure)
                className="default-Topic-lastPostInfoStyle"
                lg=""
                md="down"
                sm=""
                xl=""
                xs=""
              >
                <withStyles(AvatarPure)
                  avatarUrl="http://i70.fastpic.ru/big/2015/0628/36/ccbb1e2cb8ba8dbd379a6a12dc6b8336.jpg"
                  className=""
                  id={4}
                  size={2.5}
                />
              </withStyles(HiddenPure)>
            </withStyles(ColumnPure)>
          </withStyles(RowPure)>
        </withStyles(ColumnPure)>
      </withStyles(RowPure)>
    </withStyles(ColumnPure)>
  </withStyles(RowPure)>
  <div
    className="default-Topic-separator"
  />
</div>
`;

exports[`TopicPure component match expected snapshot without parent forum information 1`] = `
<div
  className="default-Topic-wrapper"
>
  <withStyles(RowPure)
    className=""
    reverse={false}
  >
    <withStyles(ColumnPure)
      className=""
      lg=""
      lgOffset=""
      md=""
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs={12}
      xsOffset=""
    >
      <withStyles(RowPure)
        className=""
        reverse={false}
      >
        <withStyles(ColumnPure)
          className=""
          lg={9}
          lgOffset=""
          md={7}
          mdOffset=""
          sm=""
          smOffset=""
          xl=""
          xlOffset=""
          xs={12}
          xsOffset=""
        >
          <withStyles(RowPure)
            className="default-Topic-centerRow"
            reverse={false}
          >
            <withStyles(ColumnPure)
              className="default-Topic-titleContainer"
              lg={8}
              lgOffset=""
              md=""
              mdOffset=""
              sm=""
              smOffset=""
              xl=""
              xlOffset=""
              xs={12}
              xsOffset=""
            >
              <withStyles(RowPure)
                className=""
                reverse={false}
              >
                <withStyles(ColumnPure)
                  className=""
                  lg=""
                  lgOffset=""
                  md=""
                  mdOffset=""
                  sm=""
                  smOffset=""
                  xl=""
                  xlOffset=""
                  xs={12}
                  xsOffset=""
                >
                  <withStyles(LinkPure)
                    className="default-Topic-titleStyle"
                    style={Object {}}
                    to="/Topic/1"
                  >
                    Как деактивировать бомбу
                  </withStyles(LinkPure)>
                </withStyles(ColumnPure)>
              </withStyles(RowPure)>
              <withStyles(RowPure)
                className="default-Topic-textSmallMd"
                reverse={false}
              >
                <withStyles(ColumnPure)
                  className=""
                  lg=""
                  lgOffset=""
                  md={5}
                  mdOffset=""
                  sm=""
                  smOffset=""
                  xl=""
                  xlOffset=""
                  xs=""
                  xsOffset=""
                >
                  <withStyles(HiddenPure)
                    className="default-Topic-noWrap"
                    lg="up"
                    md=""
                    sm="down"
                    xl=""
                    xs=""
                  >
                    <Connect(Term)
                      spaceAfter={true}
                      term={
                        Object {
                          "id": 2,
                          "value": "Posts",
                        }
                      }
                      untermedPostfix=":"
                    />
                    215
                  </withStyles(HiddenPure)>
                </withStyles(ColumnPure)>
              </withStyles(RowPure)>
            </withStyles(ColumnPure)>
            <withStyles(ColumnPure)
              className="default-Topic-centerRow"
              lg={2}
              lgOffset=""
              md=""
              mdOffset=""
              sm=""
              smOffset=""
              xl=""
              xlOffset=""
              xs=""
              xsOffset=""
            >
              <withStyles(HiddenPure)
                className=""
                lg=""
                md="down"
                sm=""
                xl=""
                xs=""
              >
                215
              </withStyles(HiddenPure)>
            </withStyles(ColumnPure)>
            <withStyles(ColumnPure)
              className="default-Topic-centerRow"
              lg={2}
              lgOffset=""
              md=""
              mdOffset=""
              sm=""
              smOffset=""
              xl=""
              xlOffset=""
              xs=""
              xsOffset=""
            >
              <withStyles(HiddenPure)
                className=""
                lg=""
                md="down"
                sm=""
                xl=""
                xs=""
              >
                1315
              </withStyles(HiddenPure)>
            </withStyles(ColumnPure)>
          </withStyles(RowPure)>
        </withStyles(ColumnPure)>
        <withStyles(ColumnPure)
          className=""
          lg={3}
          lgOffset=""
          md={5}
          mdOffset=""
          sm=""
          smOffset=""
          xl=""
          xlOffset=""
          xs={12}
          xsOffset=""
        >
          <withStyles(RowPure)
            className="default-Topic-heightFull"
            reverse={false}
          >
            <withStyles(ColumnPure)
              className="default-Topic-lastPostInfoStyle"
              lg={10}
              lgOffset=""
              md=""
              mdOffset=""
              sm=""
              smOffset=""
              xl=""
              xlOffset=""
              xs={12}
              xsOffset=""
            >
              <UserName
                className=""
                color="#00AA00"
                id={4}
                name="Buba"
              />
              <withStyles(HiddenPure)
                className=""
                lg="up"
                md=""
                sm=""
                xl=""
                xs=""
              >
                <Connect(Term)
                  doNotTerm={true}
                  indentAfter={true}
                  indentBefore={true}
                  spaceAfter={true}
                  spaceBefore={true}
                  untermedPostfix="-"
                />
              </withStyles(HiddenPure)>
              <Connect(RelativeDateTime)
                relativeDateTime={2016-09-19T10:42:32.000Z}
              />
            </withStyles(ColumnPure)>
            <withStyles(ColumnPure)
              className=""
              lg={2}
              lgOffset=""
              md=""
              mdOffset=""
              sm=""
              smOffset=""
              xl=""
              xlOffset=""
              xs=""
              xsOffset=""
            >
              <withStyles(HiddenPure)
                className="default-Topic-lastPostInfoStyle"
                lg=""
                md="down"
                sm=""
                xl=""
                xs=""
              >
                <withStyles(AvatarPure)
                  avatarUrl="http://i70.fastpic.ru/big/2015/0628/36/ccbb1e2cb8ba8dbd379a6a12dc6b8336.jpg"
                  className=""
                  id={4}
                  size={2.5}
                />
              </withStyles(HiddenPure)>
            </withStyles(ColumnPure)>
          </withStyles(RowPure)>
        </withStyles(ColumnPure)>
      </withStyles(RowPure)>
    </withStyles(ColumnPure)>
  </withStyles(RowPure)>
  <div
    className="default-Topic-separator"
  />
</div>
`;
